app/models/list_import/anime_planet/row.rb

```diff
@@ -0,0 +1,154 @@
+class ListImport
+  class AnimePlanet
+    class Row
+      attr_reader :node, :type
+
+      def initialize(node, type)
+        @node = node
+        @type = type
+      end
+
+      def media
+        key = "#{type}/#{media_info[:id]}"
+
+        Mapping.lookup('animeplanet', key) ||
+          Mapping.guess(type.classify.safe_constantize, media_info)
+      end
+
+      def media_info
+        {
+          id: node.attr('data-id')&.to_i,
+          title: extract_title(tooltip),
+          show_type: extract_show_type(tooltip),
+          episode_count: extract_total_episodes(tooltip),
+          chapter_count: extract_total_chapters(tooltip)
+        }.compact
+      end
+
+      def status
+        @status ||= extract_status(tooltip)
+
+        case @status
+        when /ing\z/ then :current
+        when /\AWant/ then :planned
+        when %r{Won't} then nil
+        when 'Stalled' then :on_hold
+        when 'Dropped' then :dropped
+        when 'Watched', /Read([0-9.]+)?/ then :completed
+        end
+      end
+
+      def progress
+        if status == (:completed || :planned || nil)
+          if type == 'anime'
+            amount = media_info[:episode_count]
+          else # manga
+            amount = media_info[:chapter_count]
+          end
+        else
+          extract_progress(tooltip)
+        end
+      end
+
+      def rating
+        extract_rating(tooltip)
+      end
+
+      def reconsume_count
+        if (status == :completed && type == 'anime')
+          extract_reconsume_count(tooltip)
+        end
+      end
+
+      def volumes
+        extract_volumes(tooltip) if type == 'manga'
+      end
+
+      def data
+        %i[status progress rating reconsume_count volumes].map { |k|
+          [k, send(k)]
+        }.to_h
+      end
+
+      private
+
+      def tooltip
+        @tooltip ||= Nokogiri::HTML.fragment(
+          node.at_css('.tooltip').attr('title')
+        )
+      end
+
+      # Media Info
+      def extract_title(fragment)
```

Comment:
mainly started it because of `media_info` and it looking nice with just 1 line. Kind of just adopted to everything afterwards, can prob remove most of them.